crates/metaslang/bindings/src/lib.rs

```diff
@@ -139,6 +179,119 @@ impl<KT: KindTypes + 'static> Bindings<KT> {
                 handle: *handle,
             })
     }
+
+    fn resolve_handles(&self, handles: &[GraphHandle]) -> Vec<Definition<'_, KT>> {
+        // NOTE: the Builder ensures that all handles in the parents are
+        // either definitions or references
+        handles
+            .iter()
+            .filter_map(|handle| {
+                if self.stack_graph[*handle].is_definition() {
+                    self.to_definition(*handle)
+                } else {
+                    // TODO: what should we do if the parent reference
+                    // cannot be resolved at this point?
+                    self.to_reference(*handle)
+                        .unwrap()
+                        .jump_to_definition()
+                        .ok()
+                }
+            })
+            .collect()
+    }
+
+    pub fn lookup_definition_by_name(&self, name: &str) -> Option<Definition<'_, KT>> {
```

Comment:
`lookup_definition_by_name()` should not be in the public API. That was an oversight on my part. It's only used for testing, to lookup the definition to set the resolution context. It can, in fact, be moved to the testing modules.

`get_context()` and `set_context()` should be in the public API. They should be used to set the compilation context (ie. what contract are you compiling) so that the name resolution considers the full contract hierarchy when looking up virtual method implementations. The names can certainly be improved though.

Reply:
> It can, in fact, be moved to the testing modules.

I will leave it up to you then. Would you like me to create an issue to review/update the API methods?

> They should be used to set the compilation context (ie. what contract are you compiling) so that the name resolution considers the full contract hierarchy when looking up virtual method implementations. 

This is a a bit confusing to me. Let me start a separate thread offline. Thanks!

Reply:
> They should be used to set the compilation context (ie. what contract are you compiling) so that the name resolution considers the full contract hierarchy when looking up virtual method implementations. 

This is a a bit confusing to me. Let me start a separate thread offline. Thanks!

Reply:
>I will leave it up to you then. Would you like me to create an issue to review/update the API methods?
That's a good idea. We were using the Solidity bindings rule issue #1077 to enumerate some of the pending tasks, but I think they are more related to the Bindings API. I'll create an issue and move over the relevant items from #1077.